DSharpPlus/Entities/Channel/DiscordChannel.cs

```diff
@@ -624,34 +624,70 @@ public async Task<ThreadQueryResult> ListJoinedPrivateArchivedThreadsAsync(DateT
-        ulong[] msgs = messages.Where(x => x.Channel.Id == this.Id).Select(x => x.Id).ToArray();
-        if (messages == null || !msgs.Any())
+        ArgumentNullException.ThrowIfNull(messages, nameof(messages));
+
+        List<ulong> messagesList = [];
+        foreach (DiscordMessage message in messages)
         {
-            throw new ArgumentException("You need to specify at least one message to delete.");
+            if (message.ChannelId != this.Id)
+            {
+                throw new ArgumentException(
+                    $"You cannot delete messages from channel {message.Channel.Name} through channel {this.Name}!.");
+            }
+            else if (message.Timestamp < DateTimeOffset.UtcNow.AddDays(-14))
+            {
+                throw new ArgumentException("You can only delete messages that are less than 14 days old.");
+            }
+
+            messagesList.Add(message.Id);
         }
 
-        if (msgs.Count() < 2)
+        if (messagesList.Count == 0)
+        {
+            throw new ArgumentException("You need to specify at least one message to delete.");
+        }
+        else if (messagesList.Count == 1)
+        {
+            await this.Discord.ApiClient.DeleteMessageAsync(this.Id, messagesList.Single(), reason);
+            return 1;
+        }
+        
+        for (int i = 0; i < messagesList.Count; i += 100)
         {
-            await this.Discord.ApiClient.DeleteMessageAsync(this.Id, msgs.Single(), reason);
-            return;
+            //first iter 0 - 99 -> 100 Messages
+            //second iter 100 - 199 -> 100 Messages
+            await this.Discord.ApiClient.DeleteMessagesAsync(this.Id,
+                messagesList[i .. Math.Min(messagesList.Count - 1, i + 99)], reason);
         }
 
-        for (int i = 0; i < msgs.Count(); i += 100)
+        return messagesList.Count;
+    }
+    
+    /// <summary>
+    /// Deletes multiple messages if they are less than 14 days old.  If they are older, none of the messages will be deleted and you will receive a <see cref="ArgumentException"/> error.
+    /// </summary>
+    /// <param name="messages">A collection of messages to delete.</param>
+    /// <param name="reason">Reason for audit logs.</param>
+    /// <returns></returns>
+    /// <exception cref="UnauthorizedException">Thrown when the client does not have the <see cref="Permissions.ManageMessages"/> permission.</exception>
+    /// <exception cref="NotFoundException">Thrown when the channel does not exist.</exception>
+    /// <exception cref="BadRequestException">Thrown when an invalid parameter was provided.</exception>
+    /// <exception cref="ServerErrorException">Thrown when Discord is unable to process the request.</exception>
+    public async Task<int> DeleteMessagesAsync(IAsyncEnumerable<DiscordMessage> messages, string reason = null)
+    {
+        List<DiscordMessage> list = new();
+        await foreach (DiscordMessage message in messages)
```

Comment:
it'd be ideal to batch them into 100s here and fire them off, so that we can start deleting while fetching. also, in that case, give the list a default capacity of 100